File: 693.Binary Number with Alternating Bits/bitwise.py
class Solution:
  def hasAlternatingBits(self, n : int) -> bool:
    num = (n >> 2) ^ n
    return (num & -num) == num
    
class Solution:
  def hasAlternatingBits(self, n : int) -> bool:
    num = (n >> 1) ^ n
    return (num + 1 & n) == 0

File: 693. Binary Number with Alternating Bits/bitwise.cpp
class Solution{
  public:
    bool hasAlternatingBits(int n){
      int num = (n >> 2) ^ n;
      return (num & -num) == num;
    }
};

class Solution{
  public:
  bool hasAlternatingBits(int n){
    long long num = (n >> 1) ^ n;
    return (num + 1 & n) == 0;
  }
};
